src/utils/style-formatter.js

```diff
@@ -78,17 +72,41 @@ export default {
 
     return styles;
   },
-  setFontSizeAndFamily({ button, Theme, layout }) {
-    const { style } = layout;
-    const text = button.firstElementChild;
-    text.textContent = style.label;
-    const fontFamily = colorUtils.getFontFamily(Theme);
-    setFontsize(text, button.clientHeight, fontFamily);
+  createLabelAndIcon({ button, Theme, style }) {
+    // text element wrapping label and icon
+    const text = document.createElement('text');
+    text.style.whiteSpace = 'nowrap';
+    text.style.fontFamily = colorUtils.getFontFamily(Theme);
+    // label
+    const textSpan = document.createElement('span');
+    textSpan.textContent = style.label;
+    style.font.style.underline && (textSpan.style.textDecoration = 'underline');
+    text.appendChild(textSpan);
+    // icon
+    if (style.icon.useIcon && style.icon.iconType !== '') {
+      const iconSpan = document.createElement('span');
+      iconSpan.style.textDecoration = 'none';
+      iconSpan.style.fontSize = 'inherit';
+      iconSpan.setAttribute('class', `lui-icon lui-icon--${style.icon.iconType}`);
```

Comment:
Using `lui-icon` and its types creates a very tight coupling between this component and LUI. The css and lui icon font would therefore be a requirement for this to work outside Sense, e.g. mashups. In addition, since the class names are very generic, different version of LUI/action-button could easily clash if changes have been made in either the lui css or its icon font.

This should ideally be configurable so that other font-icons could be used (material, font-awesome), at the very least it needs to be documented as a requirement.

Reply:
Hmm ok. Yeah I just noticed that the integration test does not render the icon, I'm assuming that lui is missing there... Yeah I saw that there were actually alternative icons for the old button. I'd say only using lui inside sense, and either disabling icons or using other font-icons outside sense. I'll talk to the other developers.

Reply:
Added a check so we only add the icon when in sense. We'll look at making it possible outside sense later.